migration/src/androidTest/kotlin/kin/sdk/migration/MigrationManagerIntegrationTest.kt

```diff
@@ -113,220 +130,290 @@ class MigrationManagerIntegrationTest {
     @LargeTest
     fun start_StartWhileInProcess_MigrationInProcessException() {
         assertFailsWith(MigrationInProcessException::class) {
-            val latch = CountDownLatch(1)
             val migrationManager = getNewMigrationManager(IKinVersionProvider {
                 // waiting so it will be in the middle of the migration process while starting another one.
-                latch.await(timeoutDurationSecondsShort, TimeUnit.SECONDS)
+                Thread.sleep(3000)
                 KinSdkVersion.OLD_KIN_SDK
             })
-            migrationManager.start(migrationManagerListener)
+            migrationManager.start(migrationManagerCallbacks)
             // waiting a bit so the started migration will be in the process.
-            Thread.sleep(2000)
-            migrationManager.start(migrationManagerListener)
-            latch.countDown()
+            Thread.sleep(1000)
+            migrationManager.start(migrationManagerCallbacks)
         }
     }
 
     @Test
     @LargeTest
     fun start_NeedToMigrate_getNewMigratedKinClient() {
+        val migrationDidStart = AtomicBoolean()
+        val isNewSdk = AtomicBoolean()
+        val isAccountAlreadyMigrated = AtomicBoolean()
+        var error: Error? = null
         val latch = CountDownLatch(1)
         val kinClient = getKinClientOnOldKinBlockchain()
-        var migrationStarted = false
         createAccountActivateAndMigrate(kinClient, object : IMigrationManagerCallbacks {
 
             override fun onMigrationStart() {
-                migrationStarted = true
+                migrationDidStart.set(true)
             }
 
             override fun onReady(kinClient: IKinClient) {
-                assertTrue(migrationStarted)
-                assertEquals(kinClient.getAccount(kinClient.accountCount - 1).getKinSdkVersion(), KinSdkVersion.NEW_KIN_SDK)
-                assertTrue(isAccountAlreadyMigrated())
+                val account = kinClient.getAccount(kinClient.accountCount - 1)
+                isNewSdk.set(account.kinSdkVersion == KinSdkVersion.NEW_KIN_SDK)
+                isAccountAlreadyMigrated.set(isAccountAlreadyMigrated(account.publicAddress))
                 latch.countDown()
             }
 
             override fun onError(e: Exception) {
-                fail("not supposed to reach onError with this exception: $e")
+                error = Error(e)
             }
 
         })
         assertTrue(latch.await(timeoutDurationSecondsLong, TimeUnit.SECONDS))
+        assertTrue(migrationDidStart.get())
```

Comment:
use `assertThat` to get more info in case of failure

Reply:
it is a big change as all the asserts are assertTrue.
Will do it later in a branch dedicated only to integration and unit tests